### PR TITLE
[Kingston] Purchase of more sacks

### PR DESCRIPTION
**Pull request by dracos**
Fixes https://github.com/mysociety/societyworks/issues/3021 [skip changelog]
I did it as part of the garden 'modify' flow, which mostly worked, if statement a bit big in the controller maybe but couldn't think of anything much better.

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Waste.pm [hunk 1]
```diff
@@ -1004,36 +1004,44 @@ sub garden_modify : Chained('garden_setup') : Args(0) {
...
     }
 
-    my $pick = $c->get_param('task') || '';
-    if ($pick eq 'cancel') {
-        $c->res->redirect('/waste/' . $c->stash->{property}{id} . '/garden_cancel');
-        $c->detach;
-    }
+    if ($c->stash->{garden_sacks}) {
+        my $payment_method = 'credit_card';
+        $c->forward('check_if_staff_can_pay', [ $payment_method ]); # Should always be okay here
```
Comment (struan) [inline perllib/FixMyStreet/App/Controller/Waste.pm:1009]: could this not be after the if given it's the same in both conditions? I guess it doesn't matter really.

File: t/app/controller/waste_kingston.t [hunk 2]
```diff
@@ -835,9 +827,32 @@ FixMyStreet::override_config {
...
+    subtest 'sacks, cannot modify, but can buy more' => sub {
+        set_fixed_time('2021-01-09T17:00:00Z');
+        $mech->log_in_ok($user->email);
+        $mech->get_ok('/waste/12345');
+        $mech->content_lacks('Modify your garden waste subscription');
+        $mech->content_contains('Order more garden sacks');
+        $mech->get_ok('/waste/12345/garden_modify');
+        $mech->content_contains('<span id="pro_rata_cost">41.00');
+        $mech->submit_form_ok({ with_fields => { name => 'Test McTest' } });
+        $mech->submit_form_ok({ with_fields => { tandc => 1 } });
```
Comment (struan) [inline t/app/controller/waste_kingston.t:839]: Should this maybe check to make sure there isn't a field to alter the number of sacks, just to be on the safe side.

File: perllib/FixMyStreet/App/Form/Waste/Garden/Sacks/Modify.pm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        return {};
+    },
+    finished => sub {
+        return $_[0]->wizard_finished('process_garden_modification');
+    },
+    next => 'done',
+);
+
+has_page done => (
+    title => 'Subscription amended',
```
Comment (struan) [inline perllib/FixMyStreet/App/Form/Waste/Garden/Sacks/Modify.pm:56]: This title doesn't really make sense in the context.

File: perllib/FixMyStreet/App/Controller/Waste.pm [hunk 4]
```diff
@@ -20,6 +20,7 @@ use FixMyStreet::App::Form::Waste::Garden::Cancel;
 use FixMyStreet::App::Form::Waste::Garden::Renew;
 use FixMyStreet::App::Form::Waste::Garden::Sacks;
 use FixMyStreet::App::Form::Waste::Garden::Sacks::Renew;
+use FixMyStreet::App::Form::Waste::Garden::Sacks::Modify; # Getting more
```
Comment (struan) [inline perllib/FixMyStreet/App/Controller/Waste.pm:23]: Is there a reason this could not be ::Purchase which would make the purpose clearer?

**Review by struan — CHANGES_REQUESTED** [review]
Few small naming/text things but mostly looks good.

File: perllib/FixMyStreet/App/Controller/Waste.pm (hunk 1 above)
Comment (dracos) [reply perllib/FixMyStreet/App/Controller/Waste.pm:1009]: Yeah, thought it was worth keeping separate given it's not actually doing anything here.

**Review by struan — APPROVED** [review]
👍

**Pull request merged**